src/rust/engine/options/src/parse.rs

```diff
@@ -274,49 +278,188 @@ fn format_parse_error(
     ))
 }
 
-#[allow(dead_code)]
-pub(crate) fn parse_bool(value: &str) -> Result<bool, ParseError> {
```

Comment:
Previously we called these for spot-parsing in various places, or passed these around as function pointers in generic code. Instead, we now implement these via a new trait, `Parseable`, which allows the straightforward `bool::parse()` and `bool::parse_list()` instead.